consensus/core/src/commit_syncer.rs

```diff
@@ -629,30 +629,37 @@ impl<C: NetworkClient> CommitSyncer<C> {
             }
         }
 
-        // 8. Make sure fetched block (and votes) timestamps are lower than current time.
-        for block in fetched_blocks.values().chain(vote_blocks.iter()) {
-            let now_ms = inner.context.clock.timestamp_utc_ms();
-            let forward_drift = block.timestamp_ms().saturating_sub(now_ms);
-            if forward_drift == 0 {
-                continue;
-            };
-            let peer_hostname = &inner.context.committee.authority(target_authority).hostname;
-            inner
-                .context
-                .metrics
-                .node_metrics
-                .block_timestamp_drift_wait_ms
-                .with_label_values(&[peer_hostname, "commit_syncer"])
-                .inc_by(forward_drift);
-            let forward_drift = Duration::from_millis(forward_drift);
-            if forward_drift >= inner.context.parameters.max_forward_time_drift {
-                warn!(
-                    "Local clock is behind a quorum of peers: local ts {}, certified block ts {}",
-                    now_ms,
-                    block.timestamp_ms()
-                );
+        // We want to run the following checks only if the median based commit timestamp is not enabled.
```

Comment:
Since the logic still tries to have proposed block ts > ancestor ts, maybe we need to keep the existing logic?

Reply:
I believe it is still possible to have committed blocks with maliciously high timestamps which could cause us wait here for very long time. That's why I've added the alternative waiting logic here https://github.com/MystenLabs/sui/pull/21347/files#diff-ba57be4d9057362d2cbdc16ff3463160496ab35c31623accfb896797dc4a4ef8R688 .

The one thing though that the new logic does not wait on/check is the voting blocks for which again we can't trust to wait on and block the sync process here.

Reply:
This makes sense. Committed blocks cannot be filtered so they have to decided as a whole. But for vote blocks, maybe we will need to filter them by ts first.